### ISSUE TITLE
SVM training for wrinkle detection

### ISSUE THREAD

**Issue by David-Estevez** [roboticslab-uc3m/textiles#20]
This is the final step of Li's algorithm that we have to replicate.

Steps:
* [x] Label data
* [x] Extract SIFT
* [x] Save SIFT to file
* [x] Train SVM
* [x] Fix loading SVM
* [x] Get more data
* [x] Prediction
* [ ] Test predictions
* [x] Hough transform step to retrieve the wrinkle segments
* [x] [Non-maximum supression](https://en.wikipedia.org/wiki/Canny_edge_detector#Non-maximum_suppression)

Final step (bonus):
* [x] Integration of Discontinuity and Curvature scans

**Comment by David-Estevez** [comment]
I have (sadly) discovered that OpenCV 3.1.0 [has no load method for the SVM](https://github.com/opencv/opencv/issues/4969) . :sob: 

Possible solutions are:

1. This bug is fixed in the master branch, so using the latest (3.1.0-dev) version should work. The main problem with this choice is that I'm using opencv_contrib as well, and 3.1.0-dev does not exist as a git tag. This means I cannot easily sync both repos when compiling, obtaining ugly checksum errors from cmake.
2. Use [scikit-learn](http://scikit-learn.org/stable/modules/svm.html) instead. With this choice I'm introducing another dependency, and I would have to refactor the code to cope with the API differences between `cv2`  and  `sklearn`  modules

As in any case I need more data for validation, I'll do that first while thinking which is the best solution.

**Comment by David-Estevez** [comment]
I have fixed the no load method problem by placing making a `git pull` and checking out both repos to `master`.

**Comment by David-Estevez** [comment]
Prediction implemented successfully at 88810b5021230292a99803d4a0aedfa5cbd1cb3b

We need more data to confirm that it works (using cross-validation).

**Comment by David-Estevez** [comment]
Non-maximum supression was integrated with the Hough Transform implementation in [Scikit-image](http://scikit-image.org/docs/dev/api/skimage.transform.html?highlight=hough_line_peak#skimage.transform.hough_line_peaks), so the only thing left is to integrate both approaches.

**Comment by David-Estevez** [comment]
Both approaches joined at 44fe415

Everything is implemented now, but this still needs testing with actual data. I'm closing this issue and creating a new issue related to aforementioned testing.
